http4k-core/src/main/kotlin/org/http4k/lens/query.kt

```diff
@@ -8,4 +8,6 @@ typealias QueryLens<T> = Lens<Request, T>
 object Query : BiDiLensSpec<Request, String>("query", StringParam,
     LensGet { name, target -> target.queries(name).map { it ?: "" } },
     LensSet { name, values, target -> values.fold(target.removeQuery(name)) { m, next -> m.query(name, next) } }
-)
\ No newline at end of file
+)
+
+inline fun <reified T : Enum<T>> Query.enum() = map(StringBiDiMappings.enum<T>())
```

Comment:
Also note that this commit add new-lines to the end of few files. I noticed that there's a lot of files, that lack NL at EOF even though the `.editorconfig` [mandates otherwise](https://github.com/http4k/http4k/blob/master/.editorconfig#L8). I could submit another PR that fixes this globally. It might also worth it to setup some linter eg. [ktlint](https://ktlint.github.io/) to have generally consistent formatting.